pandas/io/excel/_openpyxl.py

```diff
@@ -451,3 +462,256 @@ def write_cells(self, cells, sheet_name=None, startrow=0, startcol=0,
                             xcell = wks.cell(column=col, row=row)
                             for k, v in style_kwargs.items():
                                 setattr(xcell, k, v)
+
+
+class _OpenpyxlReader(_BaseExcelReader):
+
+    def __init__(self, filepath_or_buffer):
```

Comment:
Just on quick glance this is mostly copy / paste of the xlrd implementation. With that being the case can you just move this into the base class instead?

Reply:
I think that can be done. I did make need to make a few changes to the implementation using a BytesIO buffer, so to prevent the diff from becoming too messy I'll first move the function to the base class in a separate PR.

Reply:
Looking at it more closely this is not so straightforward. I tried to work out the details here: https://github.com/pandas-dev/pandas/compare/master...tdamsma:excel-read-shared-init-to-baseclass?expand=1 and don't think the added complexity is worth it.

Reply:
I am ok with this for now; we might want to followup and try to refactor a bit after #25427 and this are merged (@tdamsma can you create an issue about this)

Reply:
so I will take my back my previous; @tdamsma can you do a pre-cursor PR which separates things (it would be straightforward as now you know exactly what needs changing).

Reply:
Ok, started work on #26233